src/identity/interaction/ControlHandler.ts

```diff
@@ -0,0 +1,42 @@
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { APPLICATION_JSON } from '../../util/ContentTypes';
+import { InternalServerError } from '../../util/errors/InternalServerError';
+import { readJsonStream } from '../../util/StreamUtil';
+import type { InteractionHandlerInput } from './InteractionHandler';
+import { InteractionHandler } from './InteractionHandler';
+import type { InteractionRoute } from './routing/InteractionRoute';
+
+const API_VERSION = '0.3';
+
+/**
+ * Adds `controls` and `apiVersion` fields to the output of its source handler.
+ * Control paths are determined by the input routes.
+ */
+export class ControlHandler extends InteractionHandler {
+  private readonly source: InteractionHandler;
+  private readonly controls: Record<string, string>;
+
+  public constructor(source: InteractionHandler, controls: Record<string, InteractionRoute>) {
+    super();
+    this.source = source;
+    this.controls = Object.fromEntries(
+      Object.entries(controls).map(([ control, route ]): [ string, string ] => [ control, route.getPath() ]),
```

Comment:
Perhaps https://lodash.com/docs/#mapValues

Reply:
I was going to complain about lodash being a big add but apparently we already have it in there anyway. 😄